### PR TITLE
Example connect-log4j.properties for testing

### PR DESCRIPTION
**Pull request by sfc-gh-japatel**
Add test which simulates this. 
https://www.confluent.io/blog/kafka-connect-improvements-in-apache-kafka-2-3/

Clients should do this on their end. But give sample and improve documentation. 

We will also add this in documentation: 

# Testing
Tested locally.

### CODE REVIEW

File: src/test/resources/log4j.properties [hunk 1]
```diff
@@ -7,4 +7,6 @@ log4j.appender.STDOUT.layout.ConversionPattern=[%t] %5p %c{1} - %m%n
 log4j.appender.file=org.apache.log4j.RollingFileAppender
 log4j.appender.file.File=sf.log
 log4j.appender.file.layout=org.apache.log4j.PatternLayout
-log4j.appender.file.layout.ConversionPattern=%d{dd-MM-yyyy HH:mm:ss} %-5p %c{1}:%L - %m%n
\ No newline at end of file
+# date with format, Thread name, log severity, print only class name({1}) with line number, "-" message to print and platform dependent "\n"
```
Comment (sfc-gh-rramachandran) [inline src/test/resources/log4j.properties:10]: Can we add a sample log line?

File: test/connect-log4j.properties [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+
+# Send the logs to a file, rolling the file at midnight local time. For example, the `File` option specifies the
+# location of the log files (e.g. ${kafka.logs.dir}/connect.log), and at midnight local time the file is closed
+# and copied in the same directory but with a filename that ends in the `DatePattern` option.
+
+log4j.appender.connectAppender=org.apache.log4j.DailyRollingFileAppender
+log4j.appender.connectAppender.DatePattern='.'yyyy-MM-dd-HH
```
Comment (sfc-gh-rramachandran) [inline test/connect-log4j.properties:30]: Out of curiosity, are parameter names case insensitive? I am seeing both camel and pascal casing

**Review by sfc-gh-rramachandran — APPROVED** [review]
LGTM!

File: src/test/resources/log4j.properties (hunk 1 above)
Comment (sfc-gh-japatel) [reply src/test/resources/log4j.properties:10]: Will do!

File: test/connect-log4j.properties (hunk 2 above)
Comment (sfc-gh-japatel) [reply test/connect-log4j.properties:30]: connectAppender is defined above. and looks like they are case sensitive.

**Review by sfc-gh-zli — APPROVED** [review]
lgtm

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalOperations.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    /* Broken records or (Failed ingestion files) or (files in internal stage for > 1 hour ) */
+    UPLOAD_FILE_TO_TABLE_STAGE,
+
+    /* Either because of a purge file since they were ingested in SF table or because they were moved to table stage */
+    REMOVE_FILE_FROM_INTERNAL_STAGE,
+
+    /* Snowpipe REST API Usage */
+    INSERT_FILES_SNOWPIPE_API,
+
```
Comment (sfc-gh-rramachandran) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalOperations.java:22]: NIT: Comments for these 2 enums?

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalOperations.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
+package com.snowflake.kafka.connector.internal;
+
+/**
+ * Enum placeholders for internal operations.
+ * Useful mostly in logging and helps figure out where retries are happening.
```
Comment (sfc-gh-rramachandran) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalOperations.java:5]: This is a great idea for debugging known issues

**Review by sfc-gh-rramachandran — APPROVED** [review]
LGTM!

File: src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java [hunk 5]
```diff
@@ -316,7 +320,7 @@ public static Object backoffAndRetry(final SnowflakeTelemetryService telemetry,
       catch (Exception e)
       {
         finalException = e;
-        LOGGER.error(e.getMessage());
+        LOGGER.debug("Retry count:{} caught an exception for operation:{} with message:{}", iteration, operation, e.getMessage());
```
Comment (sfc-gh-japatel) [inline src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java:323]: I will change this in later PR.

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalOperations.java (hunk 3 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeInternalOperations.java:22]: Ahh, I added a general one `/* Snowpipe REST API Usage */`
Can add a start and end to be more clear next time around.

File: src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java (hunk 5 above)
Comment (sfc-gh-japatel) [reply src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java:323]: Done https://github.com/snowflakedb/snowflake-kafka-connector/pull/298/commits/513bb60e4085910e0204b4d61be18c900f656627

**Pull request merged**